mllib/src/main/scala/org/apache/spark/ml/Predictor.scala

```diff
@@ -62,6 +62,39 @@ private[ml] trait PredictorParams extends Params
     }
     SchemaUtils.appendColumn(schema, $(predictionCol), DoubleType)
   }
+
+  /**
+   * Extract [[labelCol]], weightCol(if any) and [[featuresCol]] from the given dataset,
+   * and put it in an RDD with strong types.
+   */
+  protected def extractInstances(dataset: Dataset[_]): RDD[Instance] = {
+    val w = this match {
+      case p: HasWeightCol =>
+        if (isDefined(p.weightCol) && $(p.weightCol).nonEmpty) {
+          col($(p.weightCol)).cast(DoubleType)
+        } else {
+          lit(1.0)
```

Comment:
Here too do you need a weight col, if the implementation doesn't support it (and shouldn't be calling this method)? or is it different?

Reply:
It is different from the above place. Even if a ML impl supports weighting, its `weightCol` is not necessary to be set, in this case, `lit(1)` is used implictly. Current all algs supporting weighting deal with `weightCol` in this way.